fineract-accounting/src/main/java/org/apache/fineract/accounting/producttoaccountmapping/service/ProductToGLAccountMappingHelper.java

```diff
@@ -455,4 +560,39 @@ public void deleteProductToGLAccountMapping(final Long loanProductId, final Port
             this.accountMappingRepository.deleteAllInBatch(productToGLAccountMappings);
         }
     }
+
+    public void validateChargeOffMappingsInDatabase(final List<JsonObject> mappings) {
+        final List<ApiParameterError> validationErrors = new ArrayList<>();
+
+        for (JsonObject jsonObject : mappings) {
+            final Long expenseGlAccountId = this.fromApiJsonHelper
+                    .extractLongNamed(LoanProductAccountingParams.EXPENSE_GL_ACCOUNT_ID.getValue(), jsonObject);
+            final Long chargeOffReasonCodeValueId = this.fromApiJsonHelper
+                    .extractLongNamed(LoanProductAccountingParams.CHARGE_OFF_REASON_CODE_VALUE_ID.getValue(), jsonObject);
+
+            // Validation: chargeOffReasonCodeValueId must exist in the database
+            CodeValue codeValue = this.codeValueRepositoryWrapper.findOneByCodeNameAndIdWithNotFoundDetection("ChargeOffReasons",
+                    chargeOffReasonCodeValueId);
+            if (codeValue == null) {
+                validationErrors.add(ApiParameterError.parameterError("validation.msg.chargeoffreason.invalid",
+                        "Charge-off reason with ID " + chargeOffReasonCodeValueId + " does not exist",
+                        LoanProductAccountingParams.CHARGE_OFF_REASONS_TO_EXPENSE.getValue()));
+            }
+
+            // Validation: expenseGLAccountId must exist as a valid Expense GL account
+            final GLAccount glAccount = getAccountByIdAndType(LoanProductAccountingParams.CHARGE_OFF_REASONS_TO_EXPENSE.getValue(),
```

Comment:
This throws `GLAccountNotFoundException` exception if it does not found the gl account, so `glAccount` will never be null.

Reply:
Changed to reposiroty